crates/wasm-encoder/src/aliases.rs

```diff
@@ -36,10 +36,16 @@ impl AliasSection {
     }
 
     /// Define an alias that references the export of a defined instance.
-    pub fn instance_export(&mut self, instance: u32, export: crate::Export) -> &mut Self {
+    pub fn instance_export(
+        &mut self,
+        instance: u32,
+        kind: crate::ItemKind,
```

Comment:
I guess the answer to my last question is that we need the item kind without an index in some places? I kiiiiinda am leaning towards "then also add an `ItemKind` but leave the old `Export::Whatever(u32)` pattern in place for existing uses" rather than "replace all the `Export::Whatever(u32)`s with `(ItemKind, u32)` pairs". If you feel strongly otherwise though I'm happy to back down.

Reply:
Ah yeah this is where it came into play, once I changed this from `Export` to "kind plus string" I figured I'd change the other locations as well. That being said I prefer `Export::Foo(u32)` myself, so I'll go ahead and switch back and only use `ItemKind` here. While `Export<T>` is possible I figure that's probably overkill  for now.

Reply:
> That being said I prefer `Export::Foo(u32)` myself, so I'll go ahead and switch back and only use `ItemKind` here.

Great, thanks!

> While `Export<T>` is possible I figure that's probably overkill for now.

Indeed.